### PR TITLE
feat(bpmn-service): add default types for camunda

### PR DESCRIPTION
**Pull request by akshatdubeysf**
## Description

Add default types for camunda

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Checklist:

- [X] Performed a self-review of my own code
- [X] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [X] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jamescrowley321** [comment]
@akshatdubeysf Please look at this package as well - https://www.npmjs.com/package/@types/camunda-external-task-client-js. I used that as the foundation for the worker I created. We might be able to just reuse the types there or use them as a reference. I think there are some standard API responses we are missing from Camunda.

**Comment by akshatdubeysf** [comment]
@jamescrowley321 I tried to add interfaces for the APIs for [creating deployment](https://docs.camunda.org/manual/7.15/reference/rest/deployment/post-deployment/) and [execution](https://docs.camunda.org/manual/7.15/reference/rest/process-definition/post-start-process-instance/) as those are the services available in our module, the typings for workers are already available through the typings for the camunda client so I did not add them.

**Comment by jamescrowley321** [comment]
@akshatdubeysf I see. This is fine for now. I will follow up with a way we can handle the entire API at once as far as creating a connector for Camunda.

**Review by jamescrowley321 — APPROVED** [review]

File: services/bpmn-service/src/types/camunda/variable.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export interface Variable {
```
[the remark this answers is not in the document]
Comment (akshatdubeysf) [reply services/bpmn-service/src/types/camunda/variable.ts:1]: These are variables object returned in API responses - https://docs.camunda.org/manual/7.15/reference/rest/process-definition/post-start-process-instance/, I have not added types for Variables for sending in the request, as they are defined and exported by the camunda external task client library.

File: services/bpmn-service/src/types/camunda/deployement-with-definitions.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import {Link} from './types';
+
+export interface DeploymentWithDefinitions {
+  links: Link[];
+  id: string;
+  name: string;
+  source: string;
+  tenantId: string;
+  deploymentTime: string;
+  deployedProcessDefinitions: string;
```
[the remark this answers is not in the document]
Comment (akshatdubeysf) [reply services/bpmn-service/src/types/camunda/deployement-with-definitions.ts]: fixed.

**Pull request merged**